vos/vos/commands/vrm.py

```diff
@@ -1,42 +1,78 @@
 #!python
 """remove a vospace data node, fails if container node or node is locked."""
 import logging
+import six.moves.urllib.parse as urllibparse
 from ..commonparser import set_logging_level_from_args, exit_on_exception, \
     CommonParser
-from .. import vos
+from .. import vos, storage_inventory
 
-DESCRIPTION = """remove a vospace data node; fails if container node or node is locked.
+urlparse = urllibparse.urlparse
+
+DESCRIPTION = """remove a file or a vospace data node; fails if container
+node or node is locked.
 
 eg. vrm vos:/root/node   -- deletes a data node"""
 
 
+def delete_nodes(args):
+    client = vos.Client(vospace_certfile=args.certfile,
+                        vospace_token=args.token)
+    for node in args.source:
+        if not node.startswith('vos:'):
+            raise Exception(
+                '{} is not a valid VOSpace handle'.format(node))
+        if not node.endswith('/'):
+            if client.get_node(node).islink():
+                logging.info('deleting link {}'.format(node))
+                client.delete(node)
+            elif client.isfile(node):
+                logging.info('deleting {}'.format(node))
+                client.delete(node)
+        elif client.isdir(node):
+            raise Exception('{} is a directory'.format(node))
+        else:
+            raise Exception('{} is not a directory'.format(node))
+
+
+def delete_files(args):
```

Comment:
If it was me, I would create the appropriate client up front and call one delete (different implementations) on it after determining that the arg is not a directory.